tests/TestEntry.cpp

```diff
@@ -262,3 +262,167 @@ void TestEntry::testResolveRecursivePlaceholders()
     QCOMPARE(entry5->resolveMultiplePlaceholders(entry5->url()), url);
     QCOMPARE(entry5->resolveMultiplePlaceholders(entry5->title()), QString("title+/some/path+fragment+title"));
 }
+
+void TestEntry::testResolveReferencePlaceholders()
+{
+    Database db;
+    Group* root = db.rootGroup();
+
+    Entry* entry1 = new Entry();
+    entry1->setGroup(root);
+    entry1->setUuid(Uuid::random());
+    entry1->setTitle("Title1");
+    entry1->setUsername("Username1");
+    entry1->setPassword("Password1");
+    entry1->setUrl("Url1");
+    entry1->setNotes("Notes1");
+    entry1->attributes()->set("CustomAttribute1", "CustomAttributeValue1");
+
+    Group* group = new Group();
+    group->setParent(root);
+    Entry* entry2 = new Entry();
+    entry2->setGroup(root);
+    entry2->setUuid(Uuid::random());
+    entry2->setTitle("Title2");
+    entry2->setUsername("Username2");
+    entry2->setPassword("Password2");
+    entry2->setUrl("Url2");
+    entry2->setNotes("Notes2");
+    entry2->attributes()->set("CustomAttribute2", "CustomAttributeValue2");
+
+    Entry* entry3 = new Entry();
+    entry3->setGroup(root);
+    entry3->setUuid(Uuid::random());
+    entry3->setTitle("{S:AttributeTitle}");
+    entry3->setUsername("{S:AttributeUsername}");
+    entry3->setPassword("{S:AttributePassword}");
+    entry3->setUrl("{S:AttributeUrl}");
+    entry3->setNotes("{S:AttributeNotes}");
+    entry3->attributes()->set("AttributeTitle", "TitleValue");
+    entry3->attributes()->set("AttributeUsername", "UsernameValue");
+    entry3->attributes()->set("AttributePassword", "PasswordValue");
+    entry3->attributes()->set("AttributeUrl", "UrlValue");
+    entry3->attributes()->set("AttributeNotes", "NotesValue");
+
+    Entry* tstEntry = new Entry();
+    tstEntry->setGroup(root);
+    tstEntry->setUuid(Uuid::random());
+
+    QCOMPARE(tstEntry->resolveMultiplePlaceholders(QString("{REF:T@I:%1}").arg(entry1->uuid().toHex())), QString("Title1"));
```

Comment:
IDK if I like more using `QString("Title1")` or `entry1->title()`. Opinions?

Reply:
I do not have a definite answer to this question for a general case ;( 
But in this case **entry1->title()** is more preferable, because it explicitly indicates that the test value should be equal to **entry1->title()**. I will make the necessary changes.